internal/controller/accessrequest_controller.go

```diff
@@ -189,51 +220,65 @@ func (r *AccessRequestReconciler) handlePermission(ctx context.Context, ar *api.
 		return "", fmt.Errorf("error verifying if subject is allowed: %w", err)
 	}
 	if !resp.Allowed {
-		err = r.updateStatus(ctx, ar, api.DeniedStatus, resp.Message)
+		rtHash := roleTemplateHash(rt)
+		err = r.updateStatus(ctx, ar, api.DeniedStatus, resp.Message, rtHash)
 		if err != nil {
 			return "", fmt.Errorf("error updating access request status to denied: %w", err)
 		}
 		return api.DeniedStatus, nil
 	}
 
 	details := ""
-	status, err := r.grantArgoCDAccess(ctx, ar)
+	status, err := r.grantArgoCDAccess(ctx, ar, rt)
 	if err != nil {
 		details = fmt.Sprintf("Error granting Argo CD Access: %s", err)
 	}
 	// only update status if the current state is different
 	if ar.Status.RequestState != status {
-		err = r.updateStatus(ctx, ar, status, details)
+		rtHash := roleTemplateHash(rt)
+		err = r.updateStatus(ctx, ar, status, details, rtHash)
 		if err != nil {
 			return "", fmt.Errorf("error updating access request status to granted: %w", err)
 		}
 	}
 	return status, nil
 }
 
+// roleTemplateUpdated will return true if the RoleTemplate previously associated with
+// the given AccessRequest is different than the given what is defined in the given rt.
+// Will return false otherwise.
+func roleTemplateUpdated(ar *api.AccessRequest, rt *api.RoleTemplate) bool {
+	hash := roleTemplateHash(rt)
+	if ar.Status.RoleTemplateHash != hash {
+		return true
+	}
+	return false
```

Comment:
The one-liner is easier to read 😄 
```suggestion
	return ar.Status.RoleTemplateHash != roleTemplateHash(rt)
```